### PR TITLE
adds new vanilla_average strategy

### PR DESCRIPTION
**Pull request by ziofil**
New stable Fock strategy! Credit: Eli
Turn on by setting `settings.USE_VANILLA_AVERAGE = True`

### CODE REVIEW

**Review by elib20 — APPROVED** [review]
🚀

**Review by kaspernielsen96 — APPROVED** [review]

**Review by kaspernielsen96 — APPROVED** [review]

**Review by elib20 — APPROVED** [review]
Nice!

File: mrmustard/math/lattice/pivots.py [hunk 1]
```diff
@@ -53,3 +53,19 @@ def smallest_pivot(index: tuple[int, ...]) -> tuple[int, tuple[int, ...]]:
...
+    index: tuple[int, ...],
+) -> list[tuple[int, tuple[int, ...]]]:
+    r"""returns all the pivots for the given index. A pivot is a nearest neighbor
+    of the index one index lowered.
+
+    Arguments:
+        index: the index to get the pivots of.
+
+    Returns:
+        a generator of the index that was decremented and the pivot
```
[the remark this answers is not in the document]
Comment (ziofil) [reply mrmustard/math/lattice/pivots.py:69]: nope! thanks

Comment (ziofil) [inline mrmustard/math/lattice/pivots.py:69]:
```suggestion
        a list of the index that was decremented and the pivot
```

File: mrmustard/math/lattice/strategies/vanilla.py [hunk 2]
```diff
@@ -83,6 +84,61 @@ def vanilla(shape: tuple[int, ...], A, b, c) -> ComplexTensor:  # pragma: no cov
...
+        b = np.atleast_2d(b)
+        return _vanilla_average_batch(shape, A, b, c)[..., 0]
+    elif b.ndim == 2:
+        return np.moveaxis(_vanilla_average_batch(shape, A, b, c), -1, 0)
+    else:
+        raise ValueError(f"Invalid shape for b: {b.shape}. It should be 1D or 2D.")
+
+
+@njit
+def _vanilla_average_batch(shape: tuple[int, ...], A, b, c) -> ComplexTensor:
```
Comment (ziofil) [inline mrmustard/math/lattice/strategies/vanilla.py:118]:
```suggestion
def _vanilla_average_batch(shape: tuple[int, ...], A, b, c) -> ComplexTensor:  # pragma: no cover
```

**Pull request merged**